clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1965,6 +1980,19 @@ drmemtrace_client_main(client_id_t id, int argc, const char *argv[])
                (op_record_heap.get_value() || !op_record_function.get_value().empty())) {
         FATAL("Usage error: function recording is only supported for -offline\n");
     }
+    if (op_L0_warmup_refs.get_value()) {
+
+        DR_ASSERT(!op_L0D_filter.get_value());
+        DR_ASSERT(!op_L0I_filter.get_value());
+        if (op_L0D_filter.get_value() || op_L0I_filter.get_value()) {
+            FATAL("Usage error: cannot use -L0_filter with -L0_warmup.");
```

Comment:
Hmm: what if instead of an option`-L0_warmup_refs` which requires filtering both D and I we had an option `-filter_until_refs` (or `-filter_until_instrs` if we wanted instrs) which was used in conjunction with the existing `-L0{D,I}_filter` options and switched from the specified filters to full tracing after a given ref/instruction count?  This gives flexibility in the filtering done during the warmup phase, since we already have the options and code for that flexibility.

Reply:
Yes this makes sense. I think counting instructions might be useful for applications that may have a small footprint and thus never leave the warmup mode if we counted refs. I have named it `L0_filter_until_instrs` so that it sounds related to the other `L0_filter` options.